scripts/apidoc_full.sh

```diff
@@ -12,17 +12,37 @@
 
 set -eu
 
-CONFIG_H='include/mbedtls/mbedtls_config.h'
+. $(dirname "$0")/project_detection.sh
 
-if [ -r $CONFIG_H ]; then :; else
-    echo "$CONFIG_H not found" >&2
+CRYPTO_CONFIG_H='include/psa/crypto_config.h'
+if in_mbedtls_repo; then
+    CONFIG_H='include/mbedtls/mbedtls_config.h'
+    CRYPTO_CONFIG_H='tf-psa-crypto/include/psa/crypto_config.h'
+
+    if [ -r $CONFIG_H ]; then :; else
+        echo "$CONFIG_H not found" >&2
+        exit 1
+    fi
+fi
+
+if [ -r $CRYPTO_CONFIG_H ]; then :; else
+    echo "$CRYPTO_CONFIG_H not found" >&2
     exit 1
 fi
 
-CONFIG_BAK=${CONFIG_H}.bak
-cp -p $CONFIG_H $CONFIG_BAK
+CRYPTO_CONFIG_BAK=${CRYPTO_CONFIG_H}.bak
+cp -p $CRYPTO_CONFIG_H $CRYPTO_CONFIG_BAK
 
-scripts/config.py realfull
-make apidoc
+if in_mbedtls_repo; then
+    CONFIG_BAK=${CONFIG_H}.bak
+    cp -p $CONFIG_H $CONFIG_BAK
+    scripts/config.py realfull
+    make apidoc
+    mv $CONFIG_BAK $CONFIG_H
+elif in_tf_psa_crypto_repo; then
+    scripts/config.py realfull
+    cmake -DCMAKE_BUILD_TYPE:String=Check -DGEN_FILES=ON .
```

Comment:
Having played with it, it would be better to build out of tree in a temporary directory. Otherwise when we run `doxygen.sh` locally the source tree ends up with all the CMake artefacts. `mktemp` is it seems the way to create a temporary directory in bash.

Reply:
`tmp=$(mktemp -d)` creates a temporary directory in a user-configurable location (defaulting to `/tmp`) and stores the location in `$tmp`. `tmp=$(TMPDIR=/path/to/parent mktemp -d)` creates a temporary directory under `/path/to/parent`. Avoid other options to `mktemp` for portability to older Linux and \*BSD.

But do we really need a temporary directory name? We want to keep the Doxygen build products around, so why not use a fixed directory name like `doxygen/build-apidoc-full`?

Reply:
The documentation is currently generated at the root of the source tree in the `apidoc`  directory thus we do not need to keep the build tree. A fixed build directory in the source tree would not really improve things regarding not modifying the build tree thus not sure it would be an improvement.

Reply:
A fixed build directory is slightly easier to program, and is easier to inspect and clean if an error occurs.

Reply:
okay the main thing I wanted to avoid here was to build in tree. I am fine with a build directory `doxygen/build-apidoc-full`.